lib/mayaUsd/render/MaterialXGenOgsXml/Nodes/TexcoordNodeMaya.cpp

```diff
@@ -0,0 +1,80 @@
+//
+// Copyright Contributors to the MaterialX Project
+// SPDX-License-Identifier: Apache-2.0
+//
```

Comment:
Copied from MaterialX [GeomPropValueNodeGlsl.cpp](https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/source/MaterialXGenGlsl/Nodes/GeomPropValueNodeGlsl.cpp) but adapted to read the index input of a texcoord node and convert it to USD "st" primvar naming scheme.